src/main/java/by/kukshinov/books/data/access/impl/ListBookDao.java

```diff
@@ -0,0 +1,71 @@
+package by.kukshinov.books.data.access.impl;
+
+
+import by.kukshinov.books.data.BookField;
+import by.kukshinov.books.data.access.BookDao;
+import by.kukshinov.books.data.access.DaoException;
+import by.kukshinov.books.data.factory.SearchSpecificationCreator;
+import by.kukshinov.books.data.factory.SortSpecificationCreator;
+import by.kukshinov.books.data.specefication.SearchSpecification;
+import by.kukshinov.books.data.specefication.SortBookSpecification;
+import by.kukshinov.books.model.Book;
+import by.kukshinov.books.model.ModelException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ListBookDao implements BookDao {
+
+    private final List<Book> localBookStorage = new ArrayList<>();
+
+    private final SortSpecificationCreator sortSpecificationCreator;
+
+    private final SearchSpecificationCreator searchSpecificationCreator;
+
+    public ListBookDao(
+		  SortSpecificationCreator sortSpecificationCreator,
+		  SearchSpecificationCreator searchSpecificationCreator) {
+	   this.sortSpecificationCreator = sortSpecificationCreator;
+	   this.searchSpecificationCreator = searchSpecificationCreator;
+    }
+
+
+    @Override
+    public void addBook(Book book) throws DaoException {
+	   if (localBookStorage.contains(book)) {
+		  String stringBook = book.toString();
```

Comment:
here no need for variables, just new DaoException(String.format("Book %s already exists! ", book));
toString will be called for you